### ISSUE TITLE
Add support for Api, FieldMask, SourceContext, and Type from protobuf's `.proto' files.

### ISSUE THREAD

**Issue by oldergod** [square/wire#2976]

**Comment by jianchaoxudd** [comment]
hey @oldergod ,

The field mask you in this issue is the google.protobuf.fieldmask? https://github.com/protocolbuffers/protobuf/blob/main/src/google/protobuf/field_mask.proto

**Comment by jianchaoxudd** [comment]
> Yes, all types listed are from `.proto` files within https://github.com/protocolbuffers/protobuf/tree/main/src/google/protobuf

Nice! Do you have timeline for FieldMask support at least?

**Comment by oldergod** [comment]
Not at all. They should all already be working but you'll run into troubles if you have a codebase using Wire and protoc alongside each other. Those types will be generated by both and conflict.

**Comment by jianchaoxudd** [comment]
I see. We are internally using Wire-Runtime to parse protobuf file, but there is no ProtoType.FieldMask yet.
